scalacheck/src/main/scala/pathy/scalacheck/RandomStr.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2014 - 2015 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pathy
+package scalacheck
+
+import org.scalacheck.Arbitrary
+import org.scalacheck.Gen
+
+import scalaz.Show
+
+final case class RandomStr(str: String) extends AnyVal
```

Comment:
I am wondering if we might be able to come up with a better name for this then `RandomStr`, it makes it sound too much like `Arbitrary[RandomStr]` and `Arbitrary[String]` would be the same. But the problem is I am not sure I can come up with a better name. `DefaultSeg` or `DefaultSegment`?